### PR TITLE
Implement objectForPrimaryKey() method

### LINKED ISSUE #328: Add method Realm.objectForPrimaryKey(objectType, primaryKey)
I would like to see a simple method to retrieve an object by primary key.

Ideally something similar to this would be simple to use.

`let myObject = results[ 'keyName' ];`

Or maybe what Swift developers have:

`public func objectForPrimaryKey<T: Object>(type: T.Type, key: AnyObject) -> T?`
https://realm.io/docs/swift/latest/api/Classes/Realm.html#/s:FC10RealmSwift5Realm19objectForPrimaryKeyuRdq_CS_6Object_FS0_FTMq_3keyPSs9AnyObject__GSqq__

Thank you for making realm easy to get going with on react-native!

### PR DESCRIPTION
**Pull request by appden**
Resolves #328

### CODE REVIEW

File: lib/browser/index.js [hunk 1]
```diff
@@ -106,6 +106,15 @@ export default class Realm {
         let method = util.createMethod(objectTypes.REALM, 'objects');
         return method.apply(this, [type, ...args]);
     }
+
+    objectForPrimaryKey(type, ...args) {
+        if (typeof type == 'function') {
```
Comment (alazier) [inline lib/browser/index.js]: We could just move the check for function into `typeForConstruction` to make this a one liner every time we need to do this.

File: src/js_realm.hpp [hunk 2]
```diff
@@ -207,19 +209,35 @@ class RealmClass : public ClassDefinition<T, SharedRealm> {
     }
     
     // converts constructor object or type name to type name
```
Comment (alazier) [inline src/js_realm.hpp]: this comments is no longer accurate

**Comment by appden** [comment]
@alazier I made changes according to your feedback here.

File: src/js_realm.hpp [hunk 3]
```diff
@@ -462,35 +479,45 @@ void RealmClass<T>::objects(ContextType ctx, ObjectType this_object, size_t argc
     validate_argument_count(argc, 1);
 
     SharedRealm realm = *get_internal<T, RealmClass<T>>(this_object);
-    std::string type = validated_object_type_for_value(realm, ctx, arguments[0]);
+    auto &object_schema = validated_object_schema_for_value(ctx, realm, arguments[0]);
```
Comment (alazier) [inline src/js_realm.hpp]: do you need ```auto &``` since this returns a ref?

**Comment by alazier** [comment]
everything else looks good 👍

File: src/js_realm.hpp (hunk 3 above)
Comment (appden) [inline src/js_realm.hpp]: `auto` doesn't become a reference without explicitly adding the `&`

**Pull request merged**